torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1799,7 +1806,7 @@ def _find(self, a: "sympy.Symbol") -> "sympy.Expr":
             return a
         res = self.replacements[a]
         cur_replace = {s: self._find(s) for s in res.free_symbols}
-        self.replacements[a] = self.replacements[a].xreplace(cur_replace)
+        self._set_replacement(a, self.replacements[a].xreplace(cur_replace))
```

Comment:
It's actually not obvious to me that you want to report union find path replacement the same way as new replacements discovered by unification. A union find replacement is never from the user, so I think I'm less likely to care about them.

Reply:
Maybe, but IMO it is good to have both. From what I can tell, just before adding a guard we might turn it into a replacement, which is like a equation; at that point we may or may not see a direct constant replacement. These get simplified at various points (e.g., even accessing the property `.expr` of a sym node might trigger simplification), and such steps might finally cause some constant replacement to be discovered.

If we believe that the user can simplify equations in their head to know when something has turned into a constant, then we only need the user-facing info. But it's not easy to see that always.

One way to think about this is that the "Adding shape guard..." logs are high level (and sometimes coincide with immediate specialization) and the "Specializing..." fills in everything else as fallback.